google/cloud/streaming_rpc_metadata.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STREAMING_RPC_METADATA_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STREAMING_RPC_METADATA_H
+
+#include "google/cloud/version.h"
+#include <map>
+#include <string>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * RPC request metadata.
+ *
+ * Once a RPC completes the underlying transport (gRPC or HTTP) may make some
+ * metadata attributes about the request available. You can think of these as
+ * the headers (and maybe trailers) returned by the RPC. Typically these
+ * attributes are used for troubleshooting, and are not necessary to handle the
+ * response.
+ */
+using StreamingRpcMetadata = std::multimap<std::string, std::string>;
```

Comment:
Why not `RpcMetadata`? The streaming bit seems incidental.

Reply:
Hmmm... maybe.  We don't have any way to return the metadata for non-streaming RPCs.  I did it anyway, but we can revert this commit if needed.